launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -488,11 +547,22 @@ public Process launch() throws IOException {
    * In all cases, the logger name will start with "org.apache.spark.launcher.app", to fit more
    * easily into the configuration of commonly-used logging systems.
    *
+   * If the applications are launched as a thread, the {@link SparkLauncher#redirectError} and
```

Comment:
"...is launched as a thread, ..."

Actually, better to rephrase the whole thing:

"If the application is launched in a thread, the log redirection methods are not supported, and the parent process's output and log configuration will be used."

Since these are public methods and thus show up in public documentation published on the website, please generate the javadocs and make sure there are no warnings / errors and that all the links work.